tests/lib/test_archive.py

```diff
@@ -44,6 +44,87 @@ def tgz(tmp_path: pathlib.Path) -> pathlib.Path:
     return tar
 
 
+@pytest.fixture
+def tar2(tmp_path: pathlib.Path) -> pathlib.Path:
+    a = tmp_path.joinpath("a")
+    a.write_text("")
+    b = tmp_path.joinpath("b")
+    b.write_text("")
+
+    tar = tmp_path.joinpath("tar")
+
+    with tarfile.open(tar, "w:tar") as tarf:
+        tarf.add(a, arcname="foo/v1/bin/foo")
+        tarf.add(b, arcname="foo/v1/baz")
+
+    return tar
+
+
+@pytest.fixture
+def tar3(tmp_path: pathlib.Path) -> pathlib.Path:
+    a = tmp_path.joinpath("a")
+    a.write_text("")
+    b = tmp_path.joinpath("b")
+    b.write_text("")
+
+    tar = tmp_path.joinpath("tar")
+
+    with tarfile.open(tar, "w:tar") as tarf:
+        tarf.add(a, arcname="foo/v1/bin/foo")
+        tarf.add(b, arcname="foo/bad")
+
+    return tar
+
+
+@pytest.fixture
+def tar4(tmp_path: pathlib.Path) -> pathlib.Path:
+    a = tmp_path.joinpath("a")
+    a.write_text("")
+
+    tar = tmp_path.joinpath("tar")
+
+    with tarfile.open(tar, "w:tar") as tarf:
+        tarf.add(a, arcname="/foo/bar")
+
+    return tar
+
+
+@pytest.fixture
+def tar5(tmp_path: pathlib.Path) -> pathlib.Path:
+    a = tmp_path.joinpath("a")
+    a.write_text("")
+    b = tmp_path.joinpath("b")
+    b.write_text("")
+
+    tar = tmp_path.joinpath("tar")
+
+    with tarfile.open(tar, "w:tar") as tarf:
+        tarf.add(a, arcname="foo/v1/bar")
```

Comment:
so after writing `tar6` i now realize this isn't a valid tarfile, but would it be reasonable to forgo a bunch more proper setup or should i just bite the bullet

Reply:
probably best to set it up properly so it represents a real case (and catches the problme found)